File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
     int n = nums.size();

     sort(nums.begin(), nums.end());

     int ans = nums[0];
     int freq=1;

     for(int i=1; i<n;i++){
        if(nums[i] == nums[i-1]) freq++;
        else{
            freq=1;
            ans = nums[i];
        }

        if(freq > n/2) return ans;
     }

    return ans;

    //  for(int val:nums){
    //     int freq = 0;
    //     for(int el:nums){
    //         if(val == el) freq++;
    //     }

    //     if(freq > n/2) return val;
    //  }
    // return -1;
    }

};

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let n = nums.length; 
     
     let ans =0;
     let freq=0;

    // O(n)
     for(let i=0; i<n;i++){
        if(freq == 0) ans = nums[i];
        if(ans == nums[i]) freq++;
        else {
            freq--;
        }
     }

     return ans;

    /*  

    || O(nlogn)
   // Sorting
   nums.sort((a,b) => a-b);

   // find majority element
   let freq = 1;
   let ans = nums[0];

   for(let i=1; i<n; i++ ){
    if(nums[i] == nums[i-1]){
         freq++;
    }
    else {
        freq = 1;
        ans = nums[i];
    }

   if((freq) > Math.floor(n/2)) return ans;

   }

    return ans;

    */
};